### PR TITLE
update: updated notation key command based on spec

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR updates notation key command based on the [key_spec](https://github.com/notaryproject/notation/blob/main/specs/commandline/key.md).

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: cmd/notation/key.go [hunk 1]
```diff
@@ -91,7 +87,7 @@ func keyUpdateCommand(opts *keyUpdateOpts) *cobra.Command {
 		opts = &keyUpdateOpts{}
 	}
 	command := &cobra.Command{
-		Use:     "update [name]",
+		Use:     "update [flags] <key_name>",
```
Comment (JeyJeyGao) [inline cmd/notation/key.go:90]: update --default <key_name>
Looks like `update` command is only for updating `default` key name?

Comment (patrickzheng200) [reply cmd/notation/key.go:90]: According to the [spec](https://github.com/notaryproject/notation/blob/main/specs/commandline/key.md#update-the-default-signing-key), yes.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: internal/cmd/flags.go [hunk 2]
```diff
@@ -73,9 +73,9 @@ var (
...
 
 	PflagPluginConfig = &pflag.Flag{
-		Name:      "pluginConfig",
+		Name:      "plugin-config",
 		Shorthand: "c",
-		Usage:     "list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values",
+		Usage:     "{key}={value} pairs that are passed as is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values",
 	}
 	SetPflagPluginConfig = func(fs *pflag.FlagSet, p *string) {
 		fs.StringVarP(p, PflagPluginConfig.Name, PflagPluginConfig.Shorthand, "", PflagPluginConfig.Usage)
```
Comment (shizhMSFT) [inline internal/cmd/flags.go:81]: Please see #418 for `--plugin-config` related comments.

File: cmd/notation/key.go [hunk 3]
```diff
@@ -314,3 +287,30 @@ func removeKeys(opts *keyRemoveOpts) error {
 	}
 	return nil
 }
+
+// func newX509KeyPair(opts *keyAddOpts, keyName string) (config.KeySuite, error) {
```
Comment (shizhMSFT) [inline cmd/notation/key.go:291]: Should we remove the code commented out?

File: cmd/notation/key.go [hunk 4]
```diff
@@ -148,15 +144,19 @@ func addKey(command *cobra.Command, opts *keyAddOpts) error {
 		return err
 	}
 	var key config.KeySuite
-	pluginName := opts.plugin
-	name := opts.name
+	name := strings.TrimSpace(opts.name)
```
Comment (shizhMSFT) [inline cmd/notation/key.go:147]: It is unnecessary to trim the space. Bash will do it for us ⚔️

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: internal/cmd/flags.go (hunk 2 above)
Comment (patrickzheng200) [reply internal/cmd/flags.go:81]: This fix will be implemented in #418.

File: cmd/notation/key.go [hunk 5]
```diff
@@ -62,23 +58,22 @@ func keyAddCommand(opts *keyAddOpts) *cobra.Command {
...
-		Use:   "add [key_path cert_path]",
+		Use:   "add --plugin <plugin_name> [flags] <key_name>",
 		Short: "Add key to signing key list",
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) >= 2 {
-				opts.keyPath = args[0]
-				opts.certPath = args[1]
+			if len(args) == 0 {
+				return errors.New("missing key name")
 			}
```
Comment (priteshbandi) [inline cmd/notation/key.go:66]:
```suggestion
			if len(args) != 1 {
				return errors.New("either missing key name or unnecessary parameters passed")
			}
```

File: specs/commandline/key.md [hunk 6]
```diff
@@ -30,45 +30,45 @@ Flags:
...
-  notation key add [flags] <key_name>
+  notation key add --plugin <plugin_name> [flags] <key_name>
 
 Flags:
   -d, --default                mark as default
   -h, --help                   help for add
       --id string              key id (required if --plugin is set)
   -p, --plugin string          signing plugin name
-      --plugin-config strings  {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values
+      --plugin-config strings  {key}={value} pairs that are passed as is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values
```
Comment (priteshbandi) [inline specs/commandline/key.md:40]: `plugin-config` is passed for both signing and verification plugin. [ref](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#generate-signature)
```suggestion
      --plugin-config strings  {key}={value} pairs passed as it is to a plugin, refer plugin's documentation to set appropriate values
```

File: internal/cmd/flags.go [hunk 7]
```diff
@@ -73,9 +73,9 @@ var (
 	}
 
 	PflagPluginConfig = &pflag.Flag{
-		Name:      "pluginConfig",
+		Name:      "plugin-config",
 		Shorthand: "c",
-		Usage:     "list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values",
+		Usage:     "{key}={value} pairs that are passed as is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values",
```
Comment (priteshbandi) [inline internal/cmd/flags.go:78]:
```suggestion
		Usage:     "{key}={value} pairs passed as it is to a plugin, refer plugin's documentation to set appropriate values",
```

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**